lib/Serialization/ModuleFile.cpp

```diff
@@ -990,9 +990,11 @@ void ModuleFile::collectBasicSourceFileInfo(
 
     BasicSourceFileInfo info;
     info.FilePath = filePath;
-    if (auto fingerprint = Fingerprint::fromString(fpStr))
-      info.InterfaceHashIncludingTypeMembers = fingerprint.getValue();
-    else {
+    if (auto fingerprint = Fingerprint::fromString(fpStr)) {
+      // Only the hash without type members is stored in a module file
+      // because that's what's needed for incremental imports.
+      info.InterfaceHashWithoutTypeMembers = fingerprint.getValue();
```

Comment:
This is wrong. `ModuleFileSharedCore.SourceFileListData` is a list of `BasicSourceFileInfo` that is serialized in https://github.com/apple/swift/blob/88d3facc9fa62c40af0df1f2ccb77971529f0992/lib/Serialization/SerializeDoc.cpp#L805-L821
This `fingerprint` is `InterfaceHashIncludingTypeMembers`.

Reply:
Thank you very much for pointing this out! I will fix it.